wallet/how-to/connect/index.md

```diff
@@ -1,166 +1,270 @@
+    <div>
+        <div id="providerButtons"></div>
+    </div>
+`
+
+listProviders(document.querySelector<HTMLDivElement>("#providerButtons")!)
 ```
 
-```tsx title="store.tsx"
+The `querySelector` finds and returns the first HTML element that matches the CSS selector `app`,
+and sets its `innerHTML`.
+You need to include a basic HTML structure with an inner `div` to inject a list of buttons, each
+representing a detected wallet provider.
+
+You'll create the `listProviders` function next, and pass an argument which represents the `div` element.
+This function will connect to the specific provider using the
+[`eth_requestAccounts`](/wallet/reference/eth_requestaccounts) RPC method, then use `appendChild` to
+add each button within the `div` with the `id` of `providerButtons`.
+
+#### 4. Connect to wallets
+
+Create a file `providers.ts` with the following code:
+
+```ts title="providers.ts"
 declare global {
     interface WindowEventMap {
-      "eip6963:announceProvider": CustomEvent
+        "eip6963:announceProvider": CustomEvent
     }
 }
 
-let providers: EIP6963ProviderDetail[] = []
+const connectWithProvider = async (wallet: EIP6963AnnounceProviderEvent["detail"]) => {
+    try {
+        await wallet.provider
+            .request({ method: "eth_requestAccounts" })
+    } catch (error) {
+        console.error("Failed to connect to provider:", error)
+    }
+}
 
-export const store = {
-    value: ()=>providers,
-    subscribe: (callback: ()=>void)=>{
-        function onAnnouncement(event: EIP6963AnnounceProviderEvent){
-            if(providers.map(p => p.info.uuid).includes(event.detail.info.uuid)) return
-            providers = [...providers, event.detail]
-            callback()
+export function listProviders(element: HTMLDivElement) {
+    window.addEventListener("eip6963:announceProvider",
+        (event: EIP6963AnnounceProviderEvent) => {
+            const button = document.createElement("button")
+        
+            button.innerHTML = `
+                <img src="${event.detail.info.icon}" alt="${event.detail.info.name}" />
+                <div>${event.detail.info.name}</div>
+            `
+        
+            button.onclick = () => connectWithProvider(event.detail)
+            element.appendChild(button)
         }
-        window.addEventListener("eip6963:announceProvider", onAnnouncement);
-        window.dispatchEvent(new Event("eip6963:requestProvider"));
-    
-        return ()=>window.removeEventListener("eip6963:announceProvider", onAnnouncement)
-    }
+    )
+
+    window.dispatchEvent(new Event("eip6963:requestProvider"))
 }
 ```
 
+The `connectWithProvider` function connects the user to the selected provider using
+[`eth_requestAccounts`](/wallet/reference/eth_requestaccounts).
+
+The `wallet` object is passed as an argument to the function, indicating the argument type.
+
+The `listProviders` function uses a simplified approach (over mapping and joining an entire block of HTML).
```

Comment:
Would it be better to say "...uses a simplified approach. Instead of mapping and joining an entire block of HTML, it directly passes the..."